src/indexer.c

```diff
@@ -335,15 +335,14 @@ RedisModuleDict *QueryIndex(RedisModuleCtx *ctx,
                             QueryPredicate *index_predicate,
                             size_t predicate_count) {
     RedisModuleDict *result = NULL;
-    int shouldCreateNewDict = (predicate_count > 1);
 
     PromoteSmallestPredicateToFront(ctx, index_predicate, predicate_count);
 
     // EQ or Contains
     for (int i = 0; i < predicate_count; i++) {
         if (index_predicate[i].type == EQ || index_predicate[i].type == CONTAINS ||
             index_predicate[i].type == LIST_MATCH) {
-            result = QueryIndexPredicate(ctx, &index_predicate[i], result, shouldCreateNewDict);
+            result = QueryIndexPredicate(ctx, &index_predicate[i], result, 1);
```

Comment:
If we always call `QueryIndexPredicate` with `createResultDict == 1`, lets remove `createResultDict`.

Reply:
Agree. fixed in the latest commit.